src/coreclr/jit/fgopt.cpp

```diff
@@ -5250,26 +5297,31 @@ bool Compiler::ThreeOptLayout::RunThreeOptPass(BasicBlock* startBlock, BasicBloc
             part2Size = srcPos - dstPos;
             part3Size = 1;
 
-            currScore = getScore(blockOrder[srcPos - 1], srcBlk) + getScore(blockOrder[dstPos - 1], dstBlk);
-            newScore  = candidateEdge->getLikelyWeight() + getScore(blockOrder[dstPos - 1], srcBlk);
+            currCost = GetCost(blockOrder[srcPos - 1], srcBlk) + GetCost(blockOrder[dstPos - 1], dstBlk);
+            newCost  = GetCost(srcBlk, dstBlk) + GetCost(blockOrder[dstPos - 1], srcBlk);
 
             if (srcPos != endPos)
             {
-                currScore += getScore(srcBlk, blockOrder[srcPos + 1]);
-                newScore += getScore(blockOrder[srcPos - 1], blockOrder[srcPos + 1]);
+                currCost += GetCost(srcBlk, blockOrder[srcPos + 1]);
+                newCost += GetCost(blockOrder[srcPos - 1], blockOrder[srcPos + 1]);
+            }
+            else
+            {
+                currCost += srcBlk->bbWeight;
+                newCost += blockOrder[srcPos - 1]->bbWeight;
             }
         }
 
-        // Continue evaluating candidates if this one isn't profitable
-        if ((newScore <= currScore) || Compiler::fgProfileWeightsEqual(newScore, currScore, 0.001))
+        // Continue evaluating partitions if this one isn't profitable
+        if ((currCost <= newCost) || Compiler::fgProfileWeightsEqual(currCost, newCost, 0.001))
```

Comment:
Nit, but I think this reads better...
```suggestion
        if ((newCost >= currCost) || Compiler::fgProfileWeightsEqual(currCost, newCost, 0.001))
```